### PR TITLE
Copyedit content

### PR DESCRIPTION
**Pull request by kevinkq**
## Summary of the changes / Why this is an improvement

This PR will:
* streamline the content with the crate docs tone of voice
* eliminate content-wise redundancies
* restructure the content in a meaningful manner
* position notes and warnings where they unfold their max usability
* streamline grammar syntactic inconsistencies
* fix grammatical and syntax errors

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: docs/best-practices/crate-node.txt [hunk 1]
```diff
@@ -1,434 +1,502 @@
...
+Troubleshooting with Crate node CLI
+===================================
+This part of the CrateDB documentation is based on the `Elasticsearch
+Reference [7.2] <https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html>`_.
 
-The crate-node CLI enables you to perform certain unsafe operations
-on a node that are only possible while it is shut down. This command allows you
-to adjust the role of a node and may be able to recover some data after a
-disaster.
+The Crate node CLI enables you to form a new cluster if a CrateDB cluster
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: The purpose of the CLI is not just to form a new cluster. It is also used to change the cluster state of nodes, to be able to fix a broken cluster or nodes.

File: docs/best-practices/crate-node.txt [hunk 2]
```diff
@@ -1,434 +1,502 @@
...
+still available, you can replace a broken node by a new node on a new host.
 
-crate-node CLI
-==============
+Modifying a node
+================
+The ``crate-node`` command enables you to adjust the role of a node and may also
+be able to recover some data after a disaster.
+
+You first have to call up the node, in order to modify it. Use these arguments:
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: All the operations are performed when the node is offline.

File: docs/best-practices/crate-node.txt [hunk 3]
```diff
@@ -1,434 +1,502 @@
...
+  one or more new master-eligible nodes, or maybe an unsafely-bootstrapped
+  cluster, formed in the section
+  :ref:`conf-env-re-purposing-nodes-unsafe-bootstrapping`
 
 * Stop all remaining data nodes.
 
-* On each data node, run the ``crate-node detach-cluster`` tool as shown
-  below. Verify that the tool reported Node was successfully detached from
-  the cluster.
+* On each data node, run the ``detach-cluster`` tool as demonstrated below.
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: I prefer having the full command mentioned such as ``crate-node detach-cluster``.

File: docs/best-practices/crate-node.txt [hunk 4]
```diff
@@ -1,434 +1,502 @@
...
-    of the master-eligible nodes in this cluster, and you cannot restore the
-    cluster from a snapshot. This tool can cause arbitrary data loss and its
-    use should be your last resort. If you have multiple surviving master
-    eligible nodes, you should run this tool on the node with the highest
-    cluster state (term, version) pair.
-
     Do you want to proceed?
 
-    Confirm [y/N] n
+    Confirm [y/n] n
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: N is uppercase on purpose, since it is the default option when you just press enter. This is common with command line tools. So it should be ``Confirm [y/N]``

File: docs/best-practices/crate-node.txt [hunk 5]
```diff
@@ -1,434 +1,502 @@
...
-    cluster from a snapshot. This tool can cause arbitrary data loss and its
-    use should be your last resort. If you have multiple surviving master
-    eligible nodes, you should run this tool on the node with the highest
-    cluster state (term, version) pair.
-
     Do you want to proceed?
 
-    Confirm [y/N] y
-    Master node was successfully bootstrapped
+    Confirm [y/n] y
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: N is uppercase on purpose, since it is the default option when you just press enter. This is common with command line tools. So it should be Confirm [y/N]

File: docs/best-practices/crate-node.txt [hunk 6]
```diff
@@ -1,434 +1,502 @@
...
+.. code-block:: console
+
+    node_3$ ./bin/crate-node detach-cluster
 
     Do you want to proceed?
 
-    Confirm [y/N] y
-    Node was successfully detached from the cluster
\ No newline at end of file
+    Confirm [y/n] y
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: N is uppercase on purpose, since it is the default option when you just press enter. This is common with command line tools. So it should be Confirm [y/N]

File: docs/best-practices/crate-node.txt [hunk 7]
```diff
@@ -1,434 +1,502 @@
...
+    Found 2 shards in 2 tables to clean up.
+    Use -v to see list of paths and tables affected.
+    Node is being repurposed as master and no-data. Clean-up of shard data will
+    be performed.
 
     Do you want to proceed?
 
-    Confirm [y/N] y
-    Node successfully repurposed to master and no-data.
+    Confirm [y/n] y
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: N is uppercase on purpose, since it is the default option when you just press enter. This is common with command line tools. So it should be Confirm [y/N]

File: docs/best-practices/crate-node.txt [hunk 8]
```diff
@@ -1,434 +1,502 @@
...
-    Node is being re-purposed as no-master and no-data. Clean-up of table data will be performed.
+    Found 2 tables (2 shards and 2 table metadata) to clean up.
+    Use -v to see list of paths and tables affected.
+    Node is being repurposed as no-master and no-data. Clean-up of table data
+    is performed.
 
     Do you want to proceed?
 
-    Confirm [y/N] y
+    Confirm [y/n] y
```
Comment (mkleen) [inline docs/best-practices/crate-node.txt]: N is uppercase on purpose, since it is the default option when you just press enter. This is common with command line tools. So it should be Confirm [y/N]

File: docs/best-practices/crate-node.txt [hunk 9]
```diff
@@ -1,434 +1,512 @@
...
+
+. NOTE::
+
+  You can only perform the following operations after you have shut down the
+  affected node.
+
+The ``crate-node`` command enables you to adjust the role of a node and may also
+be able to recover some data after a disaster.
+
+You first have to call up the node, in order to modify it. Use these arguments:
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: if my understanding of this section is accurate, then I think it needs rewriting a little to make it clear that (a) this is a command synopsis, and that the list afterward is a summary of what is possible (which you will then go on to explain in more detail in the following subsections)

File: docs/best-practices/crate-node.txt [hunk 10]
```diff
@@ -1,434 +1,512 @@
...
 Changing the role of a node
 ---------------------------
 
-Certain situations may require to change the initial type of a node. The
-``crate-node repurpose`` command can modify and delete the data stored on a  disk
-from a node.
+Certain situations may require you to change the initial type of a node by
+modifying and deleting its on-disk data.
+
+Run ``crate-node repurpose`` on a node with ``node.data: false`` and
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: are these config values provided on the command line? if so, reformat this so that it uses a full command-line example in a literal block

if not, explain the workflow here

File: docs/best-practices/crate-node.txt [hunk 11]
```diff
@@ -1,434 +1,512 @@
...
+``node.master: true`` to delete any remaining shard data on that node. Deleting
+remaining shard data on the node does not touch on the table metadata and
+cluster metadata.
+
+Run ``crate-node repurpose`` on a node with ``node.data: false`` and
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: are these config values provided on the command line? if so, reformat this so that it uses a full command-line example in a literal block

if not, explain the workflow here

File: docs/best-practices/crate-node.txt [hunk 12]
```diff
@@ -1,434 +1,512 @@
...
+``node.master: false`` to delete any remaining shard data and table
+metadata. Deleting remaining shard data and table metadata does not touch on the
+cluster metadata.
 
-The intended use is:
+The command ``crate-node repurpose`` is intended to:
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this info should be moved to the top of the section

File: docs/best-practices/crate-node.txt [hunk 13]
```diff
@@ -1,434 +1,512 @@
...
 
 * Stop the node
 
-* Update crate.yml by setting node.master and node.data as desired.
+* Update the `crate.yml config file` by setting ``node.master`` and
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: ``crate.yml`` (here and elsewhere)

File: docs/best-practices/crate-node.txt [hunk 14]
```diff
@@ -1,434 +1,512 @@
 .. highlight:: psql
 .. _recover:
+.. _conf-env:
```
[the remark this answers is not in the document]
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: i've exchanged conf-env with crate-node-cli where it occurred.

File: docs/best-practices/crate-node.txt [hunk 15]
```diff
@@ -1,434 +1,512 @@
...
-to let them join the cluster again. Unfortunately this is not always
-possible. A node might be unrepairable and does not start-up,
-so if the cluster is still available the node can be simply replaced by
-a fresh node on a new host which will replace the broken node.
+This solution includes certain unsafe operations that are only possible while a
+node is shut down.
 
-crate-node CLI
-==============
+You can temporarily shut down nodes for several reasons, e.g., maintenance or
```
[the remark this answers is not in the document]
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: I think we can cut this paragraph from the introduction because the information in it reoccurs in the corresponding sections and it isn't really needed there. Also, it is too detailed in that position.

File: docs/best-practices/crate-node.txt (hunk 10 above)
Comment (mkleen) [reply docs/best-practices/crate-node.txt]: These config values can be provided as comming line argument as well as set in the crate.yml config file. Usually you would set these values in crate.yml.

File: docs/best-practices/crate-node.txt (hunk 11 above)
Comment (mkleen) [reply docs/best-practices/crate-node.txt]: These config values can be provided as comming line argument as well as set in the crate.yml config file. Usually you would set these values in crate.yml.

File: docs/best-practices/crate-node.txt (hunk 10 above)
Comment (mkleen) [reply docs/best-practices/crate-node.txt]: I would not format it as a command-line block since the values are usually declared in crate.yml and not as a command line argument.

Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Okay, thanks for the info.

File: docs/best-practices/crate-node.txt [hunk 16]
```diff
@@ -1,434 +1,520 @@
 .. highlight:: psql
 .. _recover:
```
[the remark this answers is not in the document]
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Are we sure (can we check somehow) this label isn't a link target?

[the remark this answers is not in the document]
Comment (mechanomi) [reply docs/best-practices/crate-node.txt]: sphinx will throw an error if you remove a label that is in use

[the remark this answers is not in the document]
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Ah, cool. Thanks for letting me know. :)

File: docs/best-practices/crate-node.txt [hunk 17]
```diff
@@ -1,434 +1,482 @@
...
-  Shows minimal output.
+* ``-s, --silent``
+    Show minimal output.
 
-`-v, --verbose`
-  Shows verbose output.
+* ``-v, --verbose``
+    Show verbose output.
 
 Examples
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this section appears to be entirely duplicate content

File: docs/best-practices/crate-node.txt [hunk 18]
```diff
@@ -1,434 +1,482 @@
...
-find and remove excess shard data:
+.. NOTE::
+
+  Before you continue with the next step, you need to stop CrateDB.
+
+Run ``crate-node repurpose`` to find and remove excess shard data:
 
 .. code-block:: console
 
     node$ ./bin/crate-node repurpose
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: all shell commands should start with `sh$`

File: docs/best-practices/crate-node.txt [hunk 19]
```diff
@@ -1,434 +1,482 @@
...
+
+    This tool can cause arbitrary data loss, and its use should be your last
+    resort.
+
+* Run ``crate-node unsafe-bootstrap`` on the first remaining node
+  and answer ``n`` at the confirmation step.
 
 .. code-block:: console
 
     node_1$ ./bin/crate-node unsafe-bootstrap
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: all shell commands should start with `sh$`

please fix other instances

File: docs/best-practices/crate-node.txt [hunk 20]
```diff
@@ -1,434 +1,482 @@
...
+  hold the latest cluster metadata. Outdated metadata may make it impossible
+  to use some or all of the tables in the cluster.
 
-The sequence of operations for using this tool are as follows:
+To unsafely bootstrap the cluster, run:
 
-* Make sure you have really lost access to at least half of the master-eligible
-  nodes in the cluster, and they cannot be repaired or recovered by moving their
-  data paths to healthy hardware.
+``crate-node unsafe-bootstrap``
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this command, and the other commands like it that appear on a single line, should be formatted as code blocks

please also use the `console` language setting for all code blocks. you can set this once at the start of the file. you just need to update the `highlight:: psql` line

File: docs/best-practices/crate-node.txt (hunk 17 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Doesn't look like a duplicate to me really (Branch is up to date too).

File: docs/best-practices/crate-node.txt [hunk 21]
```diff
@@ -1,434 +1,442 @@
...
-completely different cluster metadata. There is no way to safely merge the
-metadata from two clusters together.
+    This tool can cause arbitrary data loss, and its use should be your last
+    resort.
 
-To protect against inadvertently joining the wrong cluster, each cluster creates
-a unique identifier, known as the cluster UUID, when it first starts up. Every
-node records the UUID of its cluster and refuses to join a cluster with a
-different UUID.
+* Run ``crate-node unsafe-bootstrap`` on the first remaining node
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: rephrase these sentences so that you're not duplicating the command in the prose. just intro the command with a description of what you're doing

File: docs/best-practices/crate-node.txt [hunk 22]
```diff
@@ -1,434 +1,442 @@
...
-  below. Verify that the tool reported Master node was successfully bootstrapped.
+.. NOTE::
 
-* Start this node and verify that it is elected as the master node.
+  This information identifies the node with the freshest cluster state, which
+  minimizes the quantity of data that might be lost.
 
-* Run the ``crate-node detach-cluster`` tool, described below, on every other
-  node in the cluster.
+Example
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: the order of the example/warning subsections is different for this section and the next section. does it make sense to standardize the structure here?

File: docs/best-practices/crate-node.txt [hunk 23]
```diff
@@ -1,434 +1,442 @@
...
 
-`--ordinal <Integer>``
-  If there is more than one node sharing a data path then this specifies which
-  node to target. Defaults to 0, meaning to use the first node in the data path.
+Before running the unsafe-bootstrap tool
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-`-C <KeyValuePair>`
-  Configures a setting.
+* Make entirely sure you have lost access to at least half of the
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: can you reformat this list so that you're not starting the section with a list

also the list items could be shortened a little/rephrased

same with the other big cautionary list

File: docs/best-practices/crate-node.txt [hunk 24]
```diff
@@ -1,434 +1,442 @@
...
+restore the cluster’s availability. In such a disaster, we recommend you
+build a new cluster from a recent snapshot and reimport any data that was
+ingested since that snapshot was taken.
+
+If it is not possible to recover the cluster from a recent snapshot, it may be
+possible to use the CLI tool to construct a new cluster that contains some of
+the data from the failed cluster.
+
+This part of the CrateDB documentation is based on the `Elasticsearch
+Reference [7.2]. <https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html>`_
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]:
```suggestion
Reference. <https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html>`_
```

also, pull this link out and put it at the end of the document. see other rst files for examples

File: docs/best-practices/crate-node.txt [hunk 25]
```diff
@@ -1,434 +1,442 @@
...
+.. NOTE::
 
-The following commands provide a various operations to modify and recover nodes:
+  You can only perform the following operations on an affected node after you
+  have shut it down.
 
-* ``crate-node repurpose`` can be used to delete unwanted data from a node if it
-  used to be a data node or a master-eligible node but has been repurposed not to
-  have one or other of these roles.
+The command ``crate-node repurpose`` is intended to:
```
[the remark this answers is not in the document]
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: A) This sentence: —The command ``crate-node repurpose`` is intended to:— was misleading due to poor rewriting. I’ve corrected that. We instead outline the overall procedure of changing the role of a node at this point.

B) Step three of the description of the procedure shows that the command is run on a node.

C) Taking that into account I implemented link request and amended the prose so that the reader gets this bit's meaning right.

D) I rephrased both the paras below and included proper examples of the command in use.

*Source-content:*

Changing the role of a node

There may be situations where you want to repurpose a node without following the proper repurposing processes. The elasticsearch-node repurpose tool allows you to delete any excess on-disk data and start a node after repurposing it.

The intended use is:

Stop the node
Update elasticsearch.yml by setting node.master and node.data as desired.
Run elasticsearch-node repurpose on the node
Start the node

If you run elasticsearch-node repurpose on a node with node.data: false and node.master: true then it will delete any remaining shard data on that node, but it will leave the index and cluster metadata alone. If you run elasticsearch-node repurpose on a node with node.data: false and node.master: false then it will delete any remaining shard data and index metadata, but it will leave the cluster metadata alone.

File: docs/best-practices/crate-node.txt (hunk 22 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: I standardized the entire document. Also I added new subheadings to the "repurpose" command section. I exchanged the word "tool" for "command" to streamline the doc. with the crate-command docs in the reference. I have rewritten many paragraphs,  structured notes and warnings and streamlined the syntax for for all admonitions.

File: docs/best-practices/crate-node.txt [hunk 26]
```diff
@@ -1,434 +1,533 @@
-.. highlight:: psql
-.. _recover:
+.. highlight:: console
+.. _crate-node-cli:
 
-====================================
-Troubleshooting with Crate-Node CLI
-====================================
+===========================================
+Troubleshooting with the crate-node command
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]:
```suggestion
Troubleshooting with the ``crate-node`` command
```

File: docs/best-practices/crate-node.txt [hunk 27]
```diff
@@ -1,434 +1,533 @@
...
+===========================================
 
-The crate-node CLI enables you to perform certain unsafe operations
-on a node that are only possible while it is shut down. This command allows you
-to adjust the role of a node and may be able to recover some data after a
-disaster.
+This section shows you how to execute certain troubleshooting operations on
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: drop "certain"

File: docs/best-practices/crate-node.txt [hunk 28]
```diff
@@ -1,434 +1,533 @@
...
+CrateDB nodes and clusters using the ``crate-node`` command, including:
 
-If a CrateDB cluster unrecoverably looses the majority of the master eligible
-nodes, the crate-node CLI provides a way to utilize the state of the remaining
-master nodes to form a new cluster.
+* changing the states of remaining master nodes to form a new cluster,
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: list items should use sentence case

Comment (mechanomi) [reply docs/best-practices/crate-node.txt]: drop the comma, and drop the ", and", and final period from the list items

Comment (mechanomi) [reply docs/best-practices/crate-node.txt]: switch from gerunds to verbs

File: docs/best-practices/crate-node.txt [hunk 29]
```diff
@@ -1,434 +1,533 @@
...
+   :maxdepth: 1
 
-.. code-block:: console
+.. contents::
+   :local:
 
-   bin/crate-node repurpose|unsafe-bootstrap|detach-cluster
-   [--ordinal <Integer>] [-C <KeyValuePair>]
-   [-h, --help] ([-s, --silent] | [-v, --verbose])
+Changing the role of a node
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: switch from gerunds to verbs

File: docs/best-practices/crate-node.txt [hunk 30]
```diff
@@ -1,434 +1,533 @@
...
+===========================
 
+Each CrateDB data node maintains the following data on disk:
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]:
```suggestion
Each CrateDB node maintains the following data on disk:
```

File: docs/best-practices/crate-node.txt [hunk 31]
```diff
@@ -1,434 +1,533 @@
...
 
-The following commands provide a various operations to modify and recover nodes:
+* the shard data for every shard allocated to that node,
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same formatting requests for this bullet list, and other lists

File: docs/best-practices/crate-node.txt [hunk 32]
```diff
@@ -1,434 +1,533 @@
...
+* the index metadata corresponding with every shard allocated to that node, and
+* the cluster-wide metadata, such as settings and index templates.
 
-* ``crate-node repurpose`` can be used to delete unwanted data from a node if it
-  used to be a data node or a master-eligible node but has been repurposed not to
-  have one or other of these roles.
+Similarly, each CrateDB  master-eligible node maintains the following data on
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: "Additionally"

File: docs/best-practices/crate-node.txt [hunk 33]
```diff
@@ -1,434 +1,533 @@
...
-  It forces one of the nodes to form a brand-new cluster on its own, using its local
-  copy of the cluster metadata.
+* the index metadata for every index in the cluster, and
+* the cluster-wide metadata, such as settings and index templates.
 
-* ``crate-node detach-cluster`` enables you to move nodes from one cluster to another.
-  This can be used to move nodes into a new cluster created with the
-  ``crate-node unsafe-bootstap`` command. If unsafe cluster bootstrapping was not
-  possible, it also enables you to move nodes into a brand-new cluster.
+Now, the `repurpose command`_ lets you delete any of a node's shard data,
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: change tense. "When you run the repurpose command, "

in fact, start this section with an intro to the repurpose command. and then add this stuff about deleting node data as a caveat / follow-on. at the moment. we just open with some facts about node data, without indicating why the reader should care

Comment (mechanomi) [reply docs/best-practices/crate-node.txt]: try to avoid possessive's with nouns like "node"

File: docs/best-practices/crate-node.txt [hunk 34]
```diff
@@ -1,434 +1,533 @@
...
+Before running the ``repurpose`` command
+----------------------------------------
 
-Certain situations may require to change the initial type of a node. The
-``crate-node repurpose`` command can modify and delete the data stored on a  disk
-from a node.
+Adhere to the following procedure to change the role of a node:
 
-The intended use is:
+* Step 1: Stop the node
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: format as a numbered list

File: docs/best-practices/crate-node.txt [hunk 35]
```diff
@@ -1,434 +1,533 @@
...
 
-Changing the role of a node
----------------------------
+Before running the ``repurpose`` command
+----------------------------------------
 
-Certain situations may require to change the initial type of a node. The
-``crate-node repurpose`` command can modify and delete the data stored on a  disk
-from a node.
+Adhere to the following procedure to change the role of a node:
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this is a bit wordy and indirect. can you rephrase?

File: docs/best-practices/crate-node.txt [hunk 36]
```diff
@@ -1,434 +1,533 @@
...
 
-The intended use is:
+* Step 1: Stop the node
+* Step 3: Update `crate.yml` by setting ``node.master`` and ``node.data`` as
+  desired
+* Step 4: Run ``crate-node repurpose`` on the node
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this doesn't match well with the title of this subsection

File: docs/best-practices/crate-node.txt [hunk 37]
```diff
@@ -1,434 +1,533 @@
...
+* Step 5: Restart the node
 
-* Stop the node
+Certain situations may require you to change the initial type of a node by
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: should this intro/explanation be moved to the start of this section?

File: docs/best-practices/crate-node.txt [hunk 38]
```diff
@@ -1,434 +1,533 @@
...
+modifying and deleting its on-disk data – for example, if you have
+irrecoverably lost the majority of the master-eligible nodes and you need to
+form a new cluster.
 
-* Update crate.yml by setting node.master and node.data as desired.
+Running the ``repurpose`` command
+---------------------------------
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: at the start of this section, it might be nice to explain/indicate that as there are three possible combinations of valid configuration settings, that corresponds to three repurposing scenarios

File: docs/best-practices/crate-node.txt [hunk 39]
```diff
@@ -1,434 +1,533 @@
...
 
-* Run ``crate-node repurpose`` on the node
+.. NOTE::
 
-* Start the node
+  Stop the node first, before run the ``repurpose`` command.
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: this is covered in the previous section

File: docs/best-practices/crate-node.txt [hunk 40]
```diff
@@ -1,434 +1,533 @@
...
 
-If you run ``crate-node repurpose`` on a node with ``node.data: false`` and
-``node.master: false`` then it will delete any remaining shard data and table
-metadata, but it will leave the cluster metadata alone.
+To delete remaining shard data on a node, without touching on the index metadata
+and cluster metadata, first update the settings in the `crate.yml` of the node
+to:
 
-**Change the type from a data to become a master node**
+* ``node.data: false``
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: please format this as a yaml code block

(same with other examples)

File: docs/best-practices/crate-node.txt (hunk 38 above)
Comment (mechanomi) [reply docs/best-practices/crate-node.txt]: in fact, thinking about it: perhaps it makes sense to replace these subsections with a table that show all possible combinations of these two config values and states whether they're valid (three of them are) and has a prose description of why you might want to do that

that will cut down on a lot of duplication in this section

File: docs/best-practices/crate-node.txt [hunk 41]
```diff
@@ -1,434 +1,533 @@
...
 
-.. WARNING::
+Unsafely bootstrapping a cluster
+================================
 
-  Execution of this command can lead to arbitrary data loss. Only run this tool
-  if you understand and accept the possible consequences and have exhausted all
-  other possibilities for recovery of your cluster.
+Using the ``unsafe-bootstrap`` command, you can unsafely override the voting
+configuration of a cluster as if performing another cluster-bootstrapping
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: can you clarify what a voting configuration is? perhaps link to a section of the docs that defines/explains it

same with the term "cluster bootstrapping"

File: docs/best-practices/crate-node.txt [hunk 42]
```diff
@@ -1,434 +1,533 @@
...
+process. This operation is useful if there is at least one remaining
+master-eligible node, but it is not possible to restart a majority of the nodes.
 
-The sequence of operations for using this tool are as follows:
+If you have a choice of nodes on which to run the ``unsafe-bootstrap`` command,
+choose the one with the highest term. The node's terms increase with
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: can you define "term"? link to to docs, etc

how do you find this value? etc

File: docs/best-practices/crate-node.txt [hunk 43]
```diff
@@ -1,434 +1,533 @@
...
+  lost.
 
-* On each data node, run the ``crate-node detach-cluster`` tool as shown
-  below. Verify that the tool reported Node was successfully detached from
-  the cluster.
+Before running the ``unsafe-bootstrap`` command
+-----------------------------------------------
 
-* If necessary, configure each data node to discover the new cluster.
+Adhere to the following procedure before running the ``detach-cluster`` command:
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: wordy (as before)

File: docs/best-practices/crate-node.txt [hunk 44]
```diff
@@ -1,434 +1,533 @@
...
 
-* Start each data node and verify that it has joined the new cluster.
+The ``unsafe-bootstrap`` command makes sure the node used to bootstrap the
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: I'm not sure what this means. but perhaps the intro section will make it clear once links/clarification is added

File: docs/best-practices/crate-node.txt [hunk 45]
```diff
@@ -1,434 +1,533 @@
...
+    healthy hardware
 
-Parameters
-----------
+  * Choose a remaining master-eligible node to become the new elected master,
+    as described in :ref:`crate-node-repurpose-node-as-master`
 
-`repurpose`
-  Delete excess data when a node’s roles are changed.
+  * Run the ``unsafe-bootstrap`` command on the chosen node, as shown below.
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: it seems unnecessarily complex to split this information over multiple sections

can we just combine it into one walkthrough?

File: docs/best-practices/crate-node.txt [hunk 46]
```diff
@@ -1,434 +1,533 @@
...
-    Do you want to proceed?
+   Current node cluster state (term, version) pair is (4, 12)
 
-    Confirm [y/N] y
-    Node successfully repurposed to no-master and no-data.
+It then asks for confirmation before taking action.
 
-Unsafe cluster bootstrapping
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Example
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same as above re collapsing this into one walkthrough

File: docs/best-practices/crate-node.txt [hunk 47]
```diff
@@ -1,434 +1,533 @@
...
+particularly useful when the voting configuration of a node has been reset, and
+a new cluster is bootstrapped.
+
+.. WARNING::
+
+  Moving nodes between clusters can lead to arbitrary data loss. Different
+  clusters can have utterly different cluster metadata, and there is no way to
+  merge the metadata of two clusters safely.
+
+Before running the ``detach-cluster`` command
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same as above re collapsing this into one walkthrough

File: docs/best-practices/crate-node.txt [hunk 48]
```diff
@@ -1,434 +1,533 @@
...
 
-Detaching nodes from their cluster
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  This means the ``detach-cluster`` command successfully completed its job. It
+  does not say that there has not been a data loss.
 
-After unsafely bootstrapping a new cluster, run the ``crate-node detach-cluster``
-command to detach all remaining nodes from the failed cluster so they can join
-the new cluster:
+Running the ``detach-cluster`` command
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same as above re collapsing this into one walkthrough

File: docs/best-practices/crate-node.txt [hunk 49]
```diff
@@ -1,434 +1,533 @@
...
-    node_3$ ./bin/crate-node detach-cluster
+    sh$ crate-node detach-cluster
+
+The node can now join another cluster with a different UUID.
+
+After unsafely bootstrapping a cluster, you need to detach all other surviving
+nodes from their old cluster so they can join the new, unsafely-bootstrapped
+cluster. See the example below.
+
+Example
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same as above re collapsing this into one walkthrough

File: docs/best-practices/crate-node.txt [hunk 50]
```diff
@@ -1,434 +1,533 @@
...
+    Master node was successfully bootstrapped
+
+  This means that ``unsafe-bootstrap`` command successfully completed its job.
+  It does not say that there has not been a data loss.
 
-In this example, a former data node is repurposed as a dedicated master node.
-First update the node’s settings to node.master: true and node.data: false in
-its crate.yml config file. Then run the ``crate-node repurpose`` command to
-find and remove excess shard data:
+Running the ``unsafe-bootstrap`` command
```
Comment (mechanomi) [inline docs/best-practices/crate-node.txt]: same as above re collapsing this into one walkthrough

File: docs/best-practices/crate-node.txt (hunk 33 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Deleting data from a node is the command's main purpose. I took over and rephrased the elasticsearch intro now. The intro does not beat around the bush but jumps right into what the command does (delete data from nodes). What do you think?

I've nixed all possessives.

File: docs/best-practices/crate-node.txt (hunk 41 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: I'm getting the feeling that a glossary could be of good use if we don't already have one.

Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Can we add the term VOTING CONFIGURATION to the appendix? https://crate.io/docs/sql-99/en/latest/appendices/f.html#v

File: docs/best-practices/crate-node.txt (hunk 42 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Can we add the term TERM to the appendix and then point to the description there? https://crate.io/docs/sql-99/en/latest/appendices/f.html#v

File: docs/best-practices/crate-node.txt (hunk 41 above)
Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Switched the first para for a cluster bootstrapping explanation. The para does not anymore mention the voting configuration.

Comment (kevinkq) [reply docs/best-practices/crate-node.txt]: Question: Can we/do we actively use/add terms like VOTING CONFIGURATION and CLUSTER BOOTSTRAPPING to the appendix and then point to their descriptions? https://crate.io/docs/sql-99/en/latest/appendices/f.html#v

**Comment by mechanomi** [comment]
@mkleen can you take another look at this? I've made some changes and I'd like you to check it from a technical perspective, please

@kevinkq can you review also? lemme know what you think!

**Comment by mkleen** [comment]
lgtm
